common/setups/rasr/util.py

```diff
@@ -429,47 +470,239 @@ def __init__(
+            lexicon=lexicon,
+            lm=lm,
+            concurrent=concurrent,
+        )
+        self.name = name
+        # from RasrDataInput: CorpusObject, Lexicon: dict, LM: dict, Concurrency: int
         self.cart_tree = cart_tree
         self.alignments = alignments
+        self.crp = crp
+        self.am_args = am_args
+        self.feature_flow = feature_flow
         self.features = features
-        # TODO add Lexicon and/or Allophone File to read external alignment
+        self.segment_path = segment_path
+        self.allophone_file = allophone_file
+        self.acoustic_mixtures = acoustic_mixtures
+        self.feature_scorers = feature_scorers
+        self.shuffle_data = shuffle_data
+
+    def get_data_dict(self):
+        return {
+            "class": "ExternSprintDataset",
+            "sprintTrainerExecPath": "sprint-executables/nn-trainer",
+            "sprintConfigStr": "",
+            "suppress_load_seqs_print": True,
+        }
+
+    def get_crp(self):
+        if self.crp is not None:
+            return self.crp
+        crp = rasr.CommonRasrParameters()
+        rasr.crp_add_default_output(crp)
+        crp.acoustic_model_config = am.acoustic_model_config(**self.am_args)
+        rasr.crp_set_corpus(crp, self.corpus_object)
+        crp.concurrent = self.concurrent
+        crp.segment_path = self.segment_path
+
+        if self.shuffle_data:
+            crp.corpus_config.segment_order_shuffle = True
+            crp.corpus_config.segment_order_sort_by_time_length = True
+            crp.corpus_config.segment_order_sort_by_time_length_chunk_size = 384
+
+        crp.lexicon_config = rasr.RasrConfig()
+        crp.lexicon_config.file = self.lexicon["filename"]
+        crp.lexicon_config.normalize_pronunciation = False
+
+        if self.cart_tree is not None:
+            crp.acoustic_model_config.state_tying.type = "cart"
+            crp.acoustic_model_config.state_tying.file = self.cart_tree
+
+        if self.lm is not None:
+            crp.language_model_config = rasr.RasrConfig()
+            crp.language_model_config.type = self.lm["type"]
+            crp.language_model_config.file = self.lm["filename"]
+            crp.language_model_config.scale = self.lm["scale"]
+
+        if self.allophone_file is not None:
+            crp.acoustic_model_config.allophones.add_from_file = self.allophone_file
+
+        return crp
+
+
+class OggZipHdfDataInput:
+    def __init__(
+        self,
+        oggzip_files: tk.Path,
+        alignments: tk.Path,
+        context_window: Optional[Dict] = None,
+        audio: Optional[Dict] = None,
+        partition_epoch: int = 1,
+        seq_ordering: str = "laplace:226",
+        targets: Optional[str] = None,
+    ):
+        self.oggzip_files = oggzip_files
+        self.alignments = alignments
+        self.context_window = (
+            context_window
+            if context_window is not None
+            else {"classes": 1, "data": 242}
```

Comment:
I do not understand what this is about.

Reply:
the values for the OggZipHdfData are taken from a setup by Peter. So for raw waveform setups. I can remove the defaults for `audio` and `context_window`

Reply:
Yes, I think this default value should be removed.